### PR TITLE
Use CMake's CUDA language support

### PR DESCRIPTION
**Pull request by mzient**
Signed-off-by: Michal Zientkiewicz <michalz@nvidia.com>

#### Why we need this PR?
- It replaces the deprecated FindCUDA with CMake's first-class support for CUDA as a programming language
- It makes it possible to have a working device-link step

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     * Added CUDA as a language to projects
     * Removed use of FindCUDA and related CMake variables
     * Replaced the variables, when necessary, with new ones from CMake
 - Affected modules and functionalities:
     * Build system / CMake files
 - Key points relevant for the review:
     * ?
 - Validation and testing:
     * CI build
 - Documentation (including examples):
     * Updated custom plugin example

**JIRA TASK**: N/A (related to DALI-1168)

### CODE REVIEW

File: cmake/modules/FindNVJPEG.cmake [hunk 1]
```diff
@@ -31,11 +32,11 @@ find_package_handle_standard_args(NVJPEG
...
     VERSION_VAR NVJPEG_VERSION)
 
-message(${CUDA_TOOLKIT_ROOT_DIR})
+# message(${CUDA_TOOLKIT_ROOT_DIR})
 if(NVJPEG_FOUND)
   # set includes and link libs for nvJpeg
-  set(CMAKE_REQUIRED_INCLUDES ${CUDA_INCLUDE_DIRS})
-  set(CMAKE_REQUIRED_LIBRARIES ${NVJPEG_LIBRARY} "-L${CUDA_TOOLKIT_ROOT_DIR}/lib64" "-lcudart_static" "-lculibos" "dl" "-pthread" "rt")
+  set(CMAKE_REQUIRED_INCLUDES ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+  #set(CMAKE_REQUIRED_LIBRARIES ${NVJPEG_LIBRARY} "-L${CUDA_TOOLKIT_ROOT_DIR}/lib64" "-lcudart_static" "-lculibos" "dl" "-pthread" "rt")
```
Comment (JanuszL) [inline cmake/modules/FindNVJPEG.cmake]: Can be removed?

File: docs/examples/custom_operations/custom_operator/create_a_custom_operator.ipynb [hunk 2]
```diff
@@ -209,7 +209,7 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "/usr/local/lib/python3.6/dist-packages/nvidia/dali/include\n"
+      "/home/michalz/.local/lib/python3.6/site-packages/nvidia/dali/include\n"
```
Comment (JanuszL) [inline docs/examples/custom_operations/custom_operator/create_a_custom_operator.ipynb]: Revert to this generic paths.

File: docs/examples/custom_operations/custom_operator/create_a_custom_operator.ipynb [hunk 3]
```diff
@@ -322,43 +324,31 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 10,
+   "execution_count": 17,
```
Comment (JanuszL) [inline docs/examples/custom_operations/custom_operator/create_a_custom_operator.ipynb]: `execution_count` looks weird now.

File: cmake/Dependencies.cmake [hunk 4]
```diff
@@ -15,14 +15,14 @@
 ##################################################################
 # CUDA Toolkit libraries (including NVJPEG)
 ##################################################################
-# Note: CUDA 8 support is unofficial.  CUDA 9 is officially supported
-find_package(CUDA 8.0 REQUIRED)
-
-include_directories(${CUDA_INCLUDE_DIRS})
-list(APPEND DALI_LIBS ${CUDA_LIBRARIES})
```
Comment (JanuszL) [inline cmake/Dependencies.cmake]: Remove from the docs `CUDA 8.0 compatibility is provided unofficially.` as well.

**Comment by JanuszL** [comment]
Please add in the commit description info that any trace of CUDA 8 support is removed.

File: cmake/Dependencies.cmake (hunk 4 above)
Comment (mzient) [reply cmake/Dependencies.cmake]: Done.

File: cmake/modules/FindNVJPEG.cmake (hunk 1 above)
Comment (mzient) [reply cmake/modules/FindNVJPEG.cmake]: In its current form, it definitely broke the build. If CI succeeds without it, I'll remove the line.

File: cmake/Dependencies.aarch64-qnx.cmake [hunk 5]
```diff
@@ -40,7 +43,7 @@ else()
   list(APPEND DALI_LIBS ${${CUDA_CUDART_LIBRARY}})
 endif()
 
-# NVIDIA NPPC library
+# NVIDIA NPP library
```
Comment (JanuszL) [inline cmake/Dependencies.aarch64-qnx.cmake]: You can fix that also in "aarch64-linux.cmake"

**Pull request merged**